mmdet/core/results/results.py

```diff
@@ -0,0 +1,346 @@
+    def numpy(self):
+        """Apply same name function to all tensors in results field."""
+        new_instance = self.new_results()
+        for k, v in self._results_field.items():
+            if isinstance(v, torch.Tensor):
+                new_instance[k] = v.cpu().numpy()
+        return new_instance
+
+    def __nice__(self):
+        repr = '\n   META INFORMATION \n'
+        for k, v in self._meta_info_field.items():
+            repr += f'{k}: {v} \n'
+        repr += '\n   PREDICTIONS \n'
+        for k, v in self._results_field.items():
+            if isinstance(v, (torch.Tensor, np.ndarray)):
+                repr += f'shape of {k}: {v.shape} \n'
+            else:
+                repr += f'{k}: {v} \n'
+        return repr
+
+    def __copy__(self):
+        # Due to the overloading of  `__getattr__`,
+        # have to customize the `__copy__`.
+        new_instance = self.__class__()
+        new_instance.__dict__.update(self.__dict__)
+        return new_instance
+
+    def __deepcopy__(self, memo=None):
+        # Due to the overloading of  `__getattr__`,
+        # have to customize the `__deepcopy__`.
+        new_instance = self.__class__()
+        for k, v in self.__dict__.items():
+            if isinstance(v, dict):
+                new_instance[k] = copy.deepcopy(v)
+        return new_instance
+
+
+class InstanceResults(Results):
+
+    def __len__(self):
+        for v in self._results_field.values():
+            return v.__len__()
+        raise NotImplementedError('This is an empty instance.')
+
+    def set(self, name, value):
+        assert isinstance(value, (torch.Tensor, np.ndarray, list)), \
+            f'Can set {type(value)}, only support' \
+            f' {(torch.Tensor, np.ndarray, list)}'
+
+        if isinstance(value, torch.Tensor) and self.device:
+            value = value.to(self.device)
+
+        for v in self._results_field.values():
+            assert len(v) == len(value), f'the length of ' \
+                                         f'values {len(value)} is ' \
+                                         f'not consistent with the length ' \
+                                         f'of this instancne {len(self)}'
+        self._results_field[name] = value
+
+    def __getitem__(self, item):
+        """
+        Args:
+            item (str:`torch.LongTensor`, obj:`torch.BoolTensor`):
+                get the corresponding values according to item.
+
+        Returns:
+            obj:`InstanceResults`: Corresponding values.
+        """
+
+        if isinstance(item, str):
+            super().__getitem__(item)
+
+        if type(item) == int:
+            if item >= len(self) or item < -len(self):
+                raise IndexError('Instances index out of range!')
+            else:
+                # keep the dimension
+                item = slice(item, None, len(self))
+
+        r_results = self.new_results()
+        if isinstance(item, (torch.LongTensor, torch.BoolTensor)):
```

Comment:
It seems that the supported type of item is not complete. We can check and refer to the cases in [mmdet3d](https://github.com/open-mmlab/mmdetection3d/blob/master/mmdet3d/core/points/base_points.py#L264) are fully supported.